### PR TITLE
RN Reanimated: fix deprecated/ removed File.exists method

### PR DESCRIPTION
**Pull request by jpcloureiro**
**Description**
react-native-reanimated 2.11.0 uses a deprecated ruby method that breaks the command `pod install` at least on ruby 2.6 onwards[1]

Problem also spotted on react-native repository[2] as well as react-native-reanimated[3]

**Screenshots/Recordings**

Screenshot 2023-01-24 at 15 18 07

This patch replaces the deprecated method with the recommended one[1]

[1] https://rubyapi.org/2.6/o/file#method-c-exist-3F
[2] https://github.com/facebook/react-native/pull/35853
[3] https://github.com/software-mansion/react-native-reanimated/pull/3919

### CODE REVIEW

File: patches/react-native-reanimated+2.11.0.patch [hunk 1]
```diff
@@ -0,0 +1,12 @@
+diff --git a/node_modules/react-native-reanimated/scripts/reanimated_utils.rb b/node_modules/react-native-reanimated/scripts/reanimated_utils.rb
+index d369e5a..3e1e6ae 100644
+--- a/node_modules/react-native-reanimated/scripts/reanimated_utils.rb
++++ b/node_modules/react-native-reanimated/scripts/reanimated_utils.rb
+@@ -1,6 +1,6 @@
+ def try_to_parse_react_native_package_json(node_modules_dir)
+   react_native_package_json_path = File.join(node_modules_dir, 'react-native/package.json')
+-  if !File.exists?(react_native_package_json_path)
++  if !File.exist?(react_native_package_json_path)
```
Comment (sethkfman) [inline patches/react-native-reanimated+2.11.0.patch:9]: Is this patched in a newer version of reanimated? I would hate for us to have to maintain this patch.

Comment (jpcloureiro) [reply patches/react-native-reanimated+2.11.0.patch:9]: It doesn't seem to be released yet.
PR https://github.com/software-mansion/react-native-reanimated/pull/3919 was merged into `main` though

Comment (sethkfman) [reply patches/react-native-reanimated+2.11.0.patch:9]: Maybe we can keep an eye out on the releases

**Review by sethkfman — COMMENTED** [review]
LGTM

**Review by sethkfman — APPROVED** [review]

**Comment by eisodev** [comment]
For anyone dropping in here who wish to stay on v2:
https://github.com/software-mansion/react-native-reanimated/releases/tag/2.14.0

Has the aforementioned fix included.

**Comment by jpcloureiro** [comment]
> For anyone dropping in here who wish to stay on v2: https://github.com/software-mansion/react-native-reanimated/releases/tag/2.14.0
> 
> Has the aforementioned fix included.

Thanks @eisodev 

Closing this in favor of https://github.com/MetaMask/metamask-mobile/pull/5729

**Comment by jpcloureiro** [comment]
Thanks @eisodev 

Closing this in favor of https://github.com/MetaMask/metamask-mobile/pull/5729

**Pull request closed without merging**